scripts/main.py

```diff
@@ -0,0 +1,232 @@
+import os
+import time 
+import pickle
+import random
+import numpy as np
+import matplotlib.pyplot as plt
+import torch
+from multiprocessing import Pool
+import vboc.model as models
+from vboc.parser import Parameters, parse_args
+from vboc.abstract import SimDynamics
+from vboc.controller import ViabilityController
+from vboc.learning import NeuralNetwork, RegressionNN, plot_viability_kernel
+
+
+def computeDataOnBorder(N_guess, test_flag=0):
+    valid_data = np.empty((0, model.nx))
+    controller.resetHorizon(N_guess)
+
+    # Randomize the initial state
+    d = np.array([random.uniform(-1, 1) for _ in range(model.nv)])
+    q = np.array([
+        random.uniform(params.q_min + model.eps, params.q_max - model.eps)
+        for _ in range(model.nq)]
+    )
+
+    # Set the initial guess
+    x_guess = np.zeros((N_guess, model.nx))
+    u_guess = np.zeros((N_guess, model.nu))
+    x_guess[:, :model.nq] = np.full((N_guess, model.nq), q)
+
+    if not test_flag:
+        # Dense sampling near the joint position bounds
+        vel_dir = random.choice([-1, 1])
+        q_init = params.q_min + model.eps if vel_dir == -1 else params.q_max - model.eps
+        q_fin = params.q_max - model.eps if vel_dir == -1 else params.q_min + model.eps
+        i = random.choice(range(model.nq))
+        d[i] = random.random() * vel_dir
+        q[i] = q_init
```

Comment:
Why are you always using an initial state  close to the joint boundaries if then you're only using the initial state to populate the dataset? This does not make sense to me.

Reply:
This part is used only if we not consider a uniform sampling (like in Algortihm 2 of the VBOC paper)